src/scripts/main.js

```diff
@@ -1,3 +1,36 @@
 'use strict';
 
-// write your code here
+const logo = document.querySelector('.logo');
+const body = document.querySelector('body');
+
+const promise1 = new Promise((resolve) => {
+  logo.addEventListener('click', () => {
+    resolve('Promise was resolved!');
+  });
+});
+
+const promise2 = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    reject(new Error('Promise was rejected!'));
+  }, 3000);
+});
+
+const showSuccessMessage = (message) => {
+  const div = document.createElement('div');
+
+  div.className = 'message';
+  div.textContent = message;
+  body.appendChild(div);
+};
+
+const showErrorMessage = (error) => {
+  const div = document.createElement('div');
+
+  div.className = 'message error-message';
+  div.textContent = error.message;
+  body.appendChild(div);
+};
+
+Promise.race([promise1, promise2])
+  .then(showSuccessMessage)
+  .catch(showErrorMessage);
```

Comment:
Using `Promise.race` will only handle the first promise that settles, whether it's resolved or rejected. If `promise1` resolves and `promise2` rejects later, the rejection will not be caught by the current `.catch(showErrorMessage)`. Consider using `Promise.allSettled` if you want to handle all promises' results or ensure that the rejection is handled appropriately.